tests/unit/fem/test_linearform_ext.cpp

```diff
@@ -0,0 +1,152 @@
+                     int vdim, int ordering, int gll, int problem, int p):
+      mesh_filename(mesh_filename),
+      mesh(Mesh::LoadFromFile(mesh_filename)),
+      dim(mesh.Dimension()), vdim(vdim), ordering(ordering), gll(gll),
+      problem(problem), p(p), q(2*p + (gll?-1:3)), fec(p, dim),
+      vfes(&mesh, &fec, vdim, ordering), mfes(&mesh, &fec, dim),
+      geom_type(vfes.GetFE(0)->GetGeomType()),
+      IntRulesGLL(0, Quadrature1D::GaussLobatto),
+      irGLL(&IntRulesGLL.Get(geom_type, q)), ir(&IntRules.Get(geom_type, q)),
+      elem_marker(), one_vec(1), dim_vec(dim), vdim_vec(vdim), vdim_dim_vec(vdim*dim),
+      cst_coeff(M_PI), dim_cst_coeff((dim_vec.Randomize(SEED), dim_vec)),
+      vdim_cst_coeff((vdim_vec.Randomize(SEED), vdim_vec)),
+      vdim_dim_cst_coeff((vdim_dim_vec.Randomize(SEED), vdim_dim_vec)),
+      vector_fct(vdim_vec_function), vdim_fct_coeff(vdim, vector_fct),
+      lf_dev(&vfes), lf_std(&vfes), lfi_dev(nullptr), lfi_std(nullptr)
+   {
+      REQUIRE(mesh.attributes.Size() == 1);
+      for (int e = 0; e < mesh.GetNE(); e++) { mesh.SetAttribute(e, e%2?1:2); }
+      mesh.SetAttributes();
+      REQUIRE(mesh.attributes.Size() == 2);
+      elem_marker.SetSize(2);
+      elem_marker[0] = 0;
+      elem_marker[1] = 1;
+
+      if (problem == DomainLF)
+      {
+         lfi_dev = new DomainLFIntegrator(cst_coeff);
+         lfi_std = new DomainLFIntegrator(cst_coeff);
+      }
+      else if (problem == DomainLFGrad)
+      {
+         lfi_dev = new DomainLFGradIntegrator(dim_cst_coeff);
+         lfi_std = new DomainLFGradIntegrator(dim_cst_coeff);
+      }
+      else if (problem == VectorDomainLF)
+      {
+         lfi_dev = new VectorDomainLFIntegrator(vdim_fct_coeff);
+         lfi_std = new VectorDomainLFIntegrator(vdim_fct_coeff);
+      }
+      else if (problem == VectorDomainLFGrad)
+      {
+         lfi_dev = new VectorDomainLFGradIntegrator(vdim_dim_cst_coeff);
+         lfi_std = new VectorDomainLFGradIntegrator(vdim_dim_cst_coeff);
+      }
+      else { REQUIRE(false); }
+
+      lfi_dev->SetIntRule(gll ? irGLL : ir);
+      lfi_std->SetIntRule(gll ? irGLL : ir);
+
+      lf_dev.AddDomainIntegrator(lfi_dev, elem_marker);
+      lf_std.AddDomainIntegrator(lfi_std, elem_marker);
+   }
+
+   void Run()
+   {
+      const bool scalar = problem == LinearFormExtTest::DomainLF ||
+                          problem == LinearFormExtTest::DomainLFGrad;
+      REQUIRE((!scalar || vdim == 1));
+
+      const bool grad = problem == LinearFormExtTest::DomainLFGrad ||
+                        problem == LinearFormExtTest::VectorDomainLFGrad;
+
+      mfem::out << "[LinearFormExt] " << dim << "D " << mesh_filename
+                << " p=" << p << " q=" << q
+                << (ordering == Ordering::byNODES ? " byNODES " : " byVDIM  ")
+                << vdim << "-"
+                << (scalar ? "Scalar" : "Vector") << (grad ? "Grad" : "Eval")
+                << std::endl;
```

Comment:
I would prefer `INFO` here instead of `mfem::out`

Also check out [`CAPTURE`](https://github.com/catchorg/Catch2/blob/v2.x/docs/logging.md#quickly-capture-value-of-variables-or-expressions) which could be even better here

Reply:
`CAPTURED`